docs/sources/concepts/configuration-syntax/_index.md

```diff
@@ -8,51 +8,81 @@ weight: 10
+loki.source.file "local_files" {
+    targets    = local.file_match.applogs.targets
+
+    // This specifies which component should process the logs next, the "link in the chain"
+    forward_to = [loki.process.add_new_label.receiver]
 }
 
-// Pattern for creating a labeled block, which the above block follows:
-BLOCK_NAME "BLOCK_LABEL" {
-  // Block body
-  IDENTIFIER = EXPRESSION // Attribute
+// Transformation: pull some data out of the log message, and turn it into a label
+loki.process "add_new_label" {
+    stage.logfmt {
+        mapping = {
+            "extracted_level" = "level",
+        }
+    }
+
+    // Add the value of "extracted_level" from the extracted map as a "level" label
+    stage.labels {
+        values = {
+            "level" = "extracted_level",
+        }
+    }
+
+    // The next link in the chain is the local_loki "receiver" (receives the telemetry)
+    forward_to = [loki.write.local_loki.receiver]
 }
 
-// Pattern for creating an unlabeled block:
-BLOCK_NAME {
-  // Block body
-  IDENTIFIER = EXPRESSION // Attribute
+// Anything that comes into this component gets written to the loki remote API
+loki.write "local_loki" {
+    endpoint {
+        url = "http://loki:3100/loki/api/v1/push"
+    }
 }
 ```
 
-{{< param "PRODUCT_NAME" >}} is designed with the following requirements in mind:
+The {{< param "PRODUCT_NAME" >}} syntax aims to reduce errors in configuration files by making configurations easier to read and write.
+{{< param "PRODUCT_NAME" >}} configurations use blocks that can be easily copied and pasted from the documentation to help you get started as quickly as possible.
 
-* _Fast_: The configuration language must be fast so the component controller can quickly evaluate changes.
-* _Simple_: The configuration language must be easy to read and write to minimize the learning curve.
-* _Debuggable_: The configuration language must give detailed information when there's a mistake in the configuration file.
+The {{< param "PRODUCT_NAME" >}} syntax is declarative, so ordering components, blocks, and attributes does not matter.
+The relationship between components determines the order of operations in the pipeline.
 
-The {{< param "PRODUCT_NAME" >}} configuration syntax is a distinct language with custom syntax and features, such as first-class functions.
+The {{< param "PRODUCT_NAME" >}} syntax uses blocks, attributes, and expressions, which are now explained, with examples.
 
-* Blocks are a group of related settings and usually represent creating a component.
-  Blocks have a name that consists of zero or more identifiers separated by `.`, an optional user label, and a body containing attributes and nested blocks.
-* Attributes appear within blocks and assign a value to a name.
-* Expressions represent a value, either literally or by referencing and combining other values.
-  You use expressions to compute a value for an attribute.
+## Blocks
```

Comment:
Are we concerned about any potential confusion now that blocks (which are composed of attributes) are explained before attributes?

Reply:
If this is a concern, but the ordering is still preferred, maybe we can add some kind of teaser to explain attributes, something like: 

> You use _Blocks_ to configure components and groups of settings called attributes.

Reply:
I think this suggestion is more clear and a good replacement for line 68.